Add plugin manager and extension locks (#14536)
jupyterlab/jupyterlab: docs/source/user/extensions.rst

     ]
   }

Managing Plugins with Plugin Manager
------------------------------------

Each JupyterLab extension is composed of one or more plugins. Plugins are also
used by JupyterLab core itself.

The Advanced Plugin Manager enables listing and disabling individual plugins,
but may have been disabled by your system administrator.

> Funcitonally, toggling an extension from the plugin manager is equivallent to

Reviewer: ```suggestion
Functionally, toggling an extension from the plugin manager is equivalent to
```

> running ``jupyter labextension enable/disable``.

Reviewer: Maybe we can make the command explicit for clarity:

```suggestion
running the ``jupyter labextension enable`` or ``jupyter labextension disable`` commands.
```

.. image:: ../images/plugin-manager-search-notebook.png
   :align: center
   :class: jp-screenshot
   :alt: An example search result in the plugin extension listing.

Locking and Unlocking Plugins
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Administrators may locked specific plugins with:

Reviewer: ```suggestion
Administrators may lock specific plugins with:
```

.. code:: bash

    jupyter labextension lock my-extension:plugin

To unlock a locked plugin:

.. code:: bash

    jupyter labextension unlock my-extension:plugin

The locked plugins appear on the pluigin list with a lock icon and cannot be enabled/disabled from the user interface:

.. image:: ../images/plugin-manager-plugin-locked.png
   :align: center
   :class: jp-screenshot
   :alt: An example row of the plugin list with `@jupyterlab/pluginmanager-extension:plugin` plugin locked.

> Managing Extensions with ``jupyter labextension``

Author: This was moved from above mostly without modification, please see comment below highlighting the place which got changed. The reasons for moving it around:
- we want users to use the Extension Manager rather than `jupyter labextension` if possible, so let's have order: Extension Manager → Plugin Manager → Command Line Interfaces
- there is a section about `jupyter labextension install` below, it was strangely separated from the rest of `jupyter labextension` section

-------------------------------------------------

The ``jupyter labextension`` command enables you to list all installed extensions,

> or disable any extension. It also bring helper commands for developers. See the

Reviewer: ```suggestion
or disable any extension. It also brings helper commands for developers. See the
```

help with ``jupyter labextension --help``.

Listing installed extensions
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

List all installed extensions with:

.. code:: bash

    jupyter labextension list

.. note::
   ``jupyter labextension`` identifies an extension by its plugin namespace

>    (the part prior to ``:``), which may be different from the name of the

Author: This was previously saying "identifies an extension by its JavaScript package name", which was inaccurate.
